zora/src/test.js

```diff
@@ -75,6 +75,10 @@ ${spec.toString()}`);
       if (error) {
         yield errorMessage({ error });
       }
+      
+      if (expectAssertions !== undefined && assertions.length !== expectAssertions) {
+        yield assertionMessage(Assert.equal(assertions.length, expectAssertions, 'expectAssertions'));
+      }
```

Comment:
I think that is not a good idea to use the regular equal operator. Creating a specific operator would be better.

Reply:
and instead of adding a message in the stream. you can simply add an assertion result at the end of the routine so it will be considered as a regular assertion:

```Javascript
const createAssertionCountResult = ({ expected, actual }) => ({
  pass: expected === actual,
  operator: 'plan',
  expected,
  actual,
  description: 'assertion count should match',
});

export const test = (description, spec, opts = defaultOptions) => {
  const { skip = false, expectAssertions = undefined } = opts;
  /* ... */

  const testRoutine = (async function () {
    try {
      const start = Date.now();
      const result = await specFn();
      executionTime = Date.now() - start;
      // add an extra assertion if needed
      if (expectAssertions) {
        onResult(
          createAssertionCountResult({
            expected: expectAssertions,
            actual: assertions.length,
          })
        );
      }
      return result;
    } catch (e) {
      error = e;
    } finally {
      done = true;
    }
  })();

  return Object.assign(testRoutine, {
    [Symbol.asyncIterator]: async function* () {

     // no need to yield a specific message here 

    /* ... */
    },
  });
};
```

and if you wish to have add the method on the assertion API instead, you could do something like:

```Javascript
export const test = (description, spec, opts = defaultOptions) => {
  const { skip = false } = opts;
  const assertions = [];
  let executionTime;
  let done = false;
  let error;
  let plan;

  /* ... */

  const specFn = skip
    ? noop
    : function zora_spec_fn() {
        return spec(
          Object.assign(assertFactory({ onResult }), {
          // extend the API here
            plan(value) {
              plan = value;
            },
          })
        );
      };

  const testRoutine = (async function () {
    try {
      const start = Date.now();
      const result = await specFn();
      executionTime = Date.now() - start;
      // add an extra assertion if needed
      if (plan) {
        onResult(
          createAssertionCountResult({
            expected: plan,
            actual: assertions.length,
          })
        );
      }
      return result;
    } catch (e) {
      error = e;
    } finally {
      done = true;
    }
  })();

  return Object.assign(testRoutine, {
    [Symbol.asyncIterator]: async function* () {
     /* ... same here */
    },
  });
};

```